Lease expiry now kills the client side of the connection too

ChrisGreenaway opened the pull request.

ljacomet commented on lease/api/src/main/java/org/terracotta/lease/LeaseExpiryConnectionKillingThread.java:
```diff
@@ -0,0 +1,60 @@
+      try {
+        Lease lease = leaseMaintainer.getCurrentLease();
+
+        if (lease instanceof LeaseImpl) {
```
After discussing with @ChrisGreenaway, this condition should be inverted to use the other `Lease` type in the check.

ChrisGreenaway commented:
I rebased and modified that instanceof check as requested. I also fixed a race in LeaseExpiryConnectionKillingThreadTest: I'd factored some common code out to the before() method; each test has its own setup to do before the thread starts; but I'd moved thread.start() to the before() - doh! Fixed now.
